src/main/java/net/openhft/chronicle/bytes/MappedBytes.java

```diff
@@ -137,7 +137,7 @@ protected void writeCheckOffset(long offset, long adding) throws BufferOverflowE
         super.writeCheckOffset(offset, adding);
     }
 
-    private void acquireNextByteStore(long offset) {
+    private synchronized void acquireNextByteStore(long offset) {
         BytesStore oldBS = bytesStore;
         try {
             bytesStore = (BytesStore) mappedFile.acquireByteStore(offset);
```

Comment:
Thread A acquires Next
Thread A assigns Next to bytesStore
Thread B assigns bytesStore(Next) to oldBS
Thread A releases oldBS
Thread B releases oldBS(Next)

Next.refCount == 0 and Next is released